### PR TITLE
Add outdated docs

### PR DESCRIPTION
**Pull request by Icemap**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [ ] master (the latest development version)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v6.0 (TiDB 6.0 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: develop/dev-guide-overview.md [hunk 1]
```diff
@@ -1,6 +1,7 @@
 ---
 title: Developer Overview
 summary: Introduce the overview of the developer guide.
+aliases: ['/tidb/dev/connectors-and-apis/','/zh/tidb/dev/connectors-and-apis/','/tidb/stable/connectors-and-apis/','/zh/tidb/stable/connectors-and-apis/','/appdev/dev/']
```
Comment (Oreoxmt) [inline develop/dev-guide-overview.md:4]: /zh 的内容是不是应该加到 docs-cn 中

**Review by Oreoxmt — COMMENTED** [review]
Rest LGTM

File: develop/dev-guide-overview.md (hunk 1 above)
Comment (Icemap) [reply develop/dev-guide-overview.md:4]: 这是旧的开发者文档中，英文文档的 aliases。
https://github.com/pingcap/docs-appdev/blob/master/en/app-dev-overview.md?plain=1#L4

File: TOC.md [hunk 2]
```diff
@@ -76,6 +76,14 @@
     - Guidelines
       - [Object Naming Convention](/develop/dev-guide-object-naming-guidelines.md)
       - [SQL Development Specifications](/develop/dev-guide-sql-development-specification.md)
+    - Outdated
```
Comment (Icemap) [inline TOC.md:79]:
```suggestion
    - Archived Docs
```

File: develop/dev-guide-overview.md (hunk 1 above)
Comment (Oreoxmt) [inline develop/dev-guide-overview.md:4]:
```suggestion
aliases: ['/tidb/dev/connectors-and-apis/','/zh/tidb/dev/connectors-and-apis/','/tidb/stable/connectors-and-apis/','/zh/tidb/stable/connectors-and-apis/','/appdev/dev/','/appdev/dev/app-dev-overview']
```

**Review by Oreoxmt — APPROVED** [review]

File: develop/dev-guide-overview.md [hunk 3]
```diff
@@ -1,6 +1,7 @@
 ---
 title: Developer Overview
 summary: Introduce the overview of the developer guide.
+aliases: ['/tidb/dev/connectors-and-apis/','/zh/tidb/dev/connectors-and-apis/','/tidb/stable/connectors-and-apis/','/zh/tidb/stable/connectors-and-apis/','/appdev/dev/','/appdev/dev/app-dev-overview']
```
Comment (Oreoxmt) [inline develop/dev-guide-overview.md:4]:
```suggestion
aliases: ['/tidb/dev/connectors-and-apis/','/tidb/stable/connectors-and-apis/','/appdev/dev/app-dev-overview']
```

**Review by qiancai — APPROVED** [review]

**Pull request merged**